wordpress-mobile/WordPress-iOS
# Modifies WPAnimatedBox to prevent endless animation

## Description
Fixes #10047, which notes high CPU utilization in the _Reader_ related to `WPAnimatedBox`.

### Root cause

1. Recently this custom view was modified to loop continuously.
1. The custom view was proliferated with the addition of "empty state" views employing `NoResultsViewController` (including _Reader_).

### Resolution
The following changes were made to address the problem:

- Simplified `-[WPAnimatedBox animate]` method.
- Exposed additional `-[WPAnimatedBox suspendAnimation]` method to end the animation loop.
- Removed `WPNoResultsView+AnimatedBox`, which was no longer in use.
- Modified `NoResultsViewController` and its _explicit_ callers to manage the animation lifecycle. It's worth noting that additional _implicit_ callers benefit from the change to `class NoResultsViewController.loadingAccessoryView()`.

### Future improvements
While the solution proposed appears to resolve the issue, it does require that future users of `WPAnimatedBox` & `NoResultsViewController` follow a convention of sorts. 

In the future, we might consider additional safeguards. For example:
- We might designate a maximum animation duration or loop count for `WPAnimatedBox`
- We might improve the handling of `accessoryView`;  we could clarify the type (currently `UIView!` or even suspend animations if a `WPAnimatedBox` is set where animations are currently in-flight.

## To test

- Checkout branch and verify that existing tests pass.
- Because this is related to CPU utilization, this will be easier to test in an Xcode simulator. To do so: run the app in a variety of scenarios (e.g., signed in, out, different accounts); exercise a variety of _No Results_ views, including the Reader; verify that CPU utilization in _Debug navigator_ and/or Instruments continues to behave.
- It's possible that I might have missed a _No Results_ code path; please let me know if you see anything!

cc: @loremattei for v10.8 visibility
